src/lodepng/lodepng.cpp

```diff
@@ -4330,7 +4322,19 @@ static unsigned lodepng_encode(unsigned char** out, size_t* outsize,
   /* color convert and compute scanline filter types */
   lodepng_info_copy(&info, &state->info_png);
   if(state->encoder.auto_convert) {
-    state->error = lodepng_auto_choose_color(&info.color, image, w, h, &state->info_raw, state->div);
+    LodePNGColorStats stats;
+    lodepng_color_stats_init(&stats);
+
+    state->error = lodepng_compute_color_stats(&stats, image, numpixels, &state->info_raw);
+    if(state->error) goto cleanup;
+    else { /*check if image is white only if no error is detected in previous function*/
+      unsigned char r = 0, g = 0, b = 0, a = 0;
+      getPixelColorRGBA8(&r, &g, &b, &a, image, 0, &state->info_raw);
+      stats.white = stats.numcolors == 1 && stats.colored == 0 && r == 255 && w > 20 && h > 20
```

Comment:
Why does this get moved?

Reply:
This reduces the number of variables passed to lodepng_compute_color_stats by one, this change also gets around calculating numpixels twice